### PR TITLE
New package: xidlehook-0.9.1

### LINKED ISSUE #27422: [Package request] xidlehook
- **Homepage**: https://github.com/jD91mZM2/xidlehook
- **Description**: xidlehook is a general-purpose replacement for xautolock. It executes a command when the computer has been idle for a specified amount of time.

(Tried to create a package myself, but running into problems with cargo-style build process. Looks like some manual handling is required in this case.)

### PR DESCRIPTION
**Pull request by xXR01I1Xx**
closes #27422

### CODE REVIEW

File: srcpkgs/xidlehook/template [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'xidlehook'
+pkgname=xidlehook
+version=0.9.1
+revision=1
+build_style=cargo
+configure_args="$(vopt_if audio '--features pulse' '--no-default-features')"
+make_install_args="--path xidlehook-daemon"
+hostmakedepends="pkg-config"
+makedepends="libxcb-devel libXScrnSaver-devel $(vopt_if audio libpulseaudio)"
+short_desc="xidlehook is a general-purpose replacement for xautolock"
```
Comment (abenson) [inline srcpkgs/xidlehook/template:10]: Could be shortened: `short_desc="General-purpose replacement for xautolock"`

File: srcpkgs/xidlehook/template [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Template file for 'xidlehook'
+pkgname=xidlehook
+version=0.9.1
+revision=1
+archs="x86_64 x86_64-musl aarch64 aarch64-musl"
```
Comment (abenson) [inline srcpkgs/xidlehook/template:5]: Why limit archs?

File: srcpkgs/xidlehook/template [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+build_style=cargo
+configure_args="$(vopt_if audio '--features pulse' '--no-default-features')"
+make_install_args="--path xidlehook-daemon"
+hostmakedepends="pkg-config python3"
+makedepends="libxcb-devel libXScrnSaver-devel $(vopt_if audio libpulseaudio)"
```
Comment (abenson) [inline srcpkgs/xidlehook/template:10]: Shouldn't this `libpulseaudio-devel`?

**Comment by Meister1593** [comment]
Can this be merged to main package list? Quite useful alternative to xautolock
Project (xidlehook) by itself has been stalled for a while though

**Pull request closed without merging**